docs/root/configuration/network_filters/forward_original_sni.rst

```diff
@@ -0,0 +1,14 @@
+.. _config_network_filters_forward_original_sni:
+
+Forward Original SNI
+=========================
+
+The `forward_original_sni` is a network filter that instructs other filters,
```

Comment:
given this explanation, I think this should be called use_downstream_sni .
The explanation also needs to be tweaked to state that when an inbound connection is using TLS, and the upstream cluster is using mTLS, this option will allow user to automatically set the SNI of upstream cluster to be the same as the downstream SNI.

Reply:
@rshriram Agreed, `used_downstream_sni` is more clear.

Regarding: 

> The explanation also needs to be tweaked to state that when an inbound connection is using TLS, and the upstream cluster is using mTLS, this option will allow user to automatically set the SNI of upstream cluster to be the same as the downstream SNI.

So why does the explanation not cover the case you are describing? If we can consider mTLS being a kind of TLS: `a network filter that instructs other filters, such as tcp_proxy, to forward the SNI value from the downstream connection to the upstream connection`